src/main/src/main/java/org/geoserver/catalog/ResourcePool.java

```diff
@@ -2167,6 +2167,11 @@ public BufferedReader readStyle(StyleInfo style) throws IOException {
         if (styleResource == null) {
             throw new IOException("No such resource: " + style.getFilename());
         }
+
+        // return an empty content if the file is missing.
+        if (styleResource.getType() == Resource.Type.UNDEFINED)
+            return new BufferedReader(new org.apache.commons.io.input.NullReader());
```

Comment:
Why not throw the same exception as above? The code path is already expecting an IOException if the resource is not available ...

Reply:
It seems this was the intention here. But since it never worked before, this was not taken into account. To do this right, it currently causes additional ugly catch/try brocks at different places. This is just a quick fix, to be able to finish the main idea: throw an exception for r.in() on missing files.

After that was finished, I can propose a more general solution here. It's the same as with [GEOS-10723](https://github.com/geoserver/geoserver/pull/6304#issuecomment-1307330343) and the functional approach. It's still on my list.

Reply:
StyleAdminPanel and SLDNamedLayerRenameHelper are the only non-community module classes that call this method and they seem to already expect the exception and have the appropriate try/catch blocks.

Reply:
If there are unit tests that depend on a bug then fixing the bug should require changing those tests.  Could you specifically list places that would require impractical changes?  The two places that I found already seem to be expecting the exception to be thrown and should probably be changed if you are fixing this in a way that doesn't throw the exception.

Reply:
To throw the exception is the right solution. But it definitely changes the behavior. This affects much more possible use cases and there are a lot of unit tests missing to verify all these cases. So, I don't want to fix this in the current PR to keep it small.

This PR has another topic: to avoid calling res.in(), and this way it preserves the original behavior (even if it seems wrong). This is currently important, since it blocks [GEOS-10705].

After all pending subtasks have been resolved, I can come back and review the Exception problem the way you proposed. But that will be another issue then.
This one is only an intermediate minimal fix.